pennylane/devices/qubit/sampling.py

```diff
@@ -307,7 +282,18 @@ def _process_single_shot(samples):
             raise e
         samples = qml.math.full((shots.total_shots, len(wires)), 0)
 
-    return _process_single_shot(samples)
+    processed_samples = []
+    for lower, upper in shots.bins():
+        if is_state_batched:
+            # Handle broadcasting
+            processed_samples.append(_process_single_shot(samples[:, lower:upper, :]))
+        else:
+            processed_samples.append(_process_single_shot(samples[lower:upper]))
```

Comment:
Do we need the second empty slice `:` in the first clause? As far as I am aware it does not do anything here?
There would be the option to do
```suggestion
        processed_samples.append(_process_single_shot(samples[..., lower:upper, :]))
```
(In this case, the trailing slice `:` makes sure that the Ellipsis `...` does not consume a dimension when there is no broadcasting, and only one if there is.)
@albi3ro might want to make the call whether we want this more compressed variant or go for more explicit, verbose code.

Reply:
Either works fine for me, as long as they both give equally correct answers.

Reply:
I opted for the more compact version as suggested by @dwierichs. I overlooked the trailing slice `:` , which caused the tests to fail. Appreciate the advice, @dwierichs!